### DISCUSSION TITLE
ColorPicker: Number Sign on Hex Input Values

### DISCUSSION THREAD

**Discussion by hzhu** [microsoft/fluentui#16008] [Q&A]
<img width="497" alt="Screen Shot 2020-11-19 at 1 55 59 PM" src="https://user-images.githubusercontent.com/1811365/99728912-021b1d80-2a6f-11eb-9095-5a54098248dd.png">

I noticed that there's no option to include a number sign in front of HEX input values, e.g. `#FF00FF`. It seemed to have been left off on purpose. I have a few questions regarding this decision:

1) Why was the number sign (#) left off of HEX input values? What drove this decision?

2) Is there a way for consumers to configure the `ColorPicker` so that the input renders HEX colors with the number sign (#)?

**Comment by ecraig12345** [comment]
The ColorPicker was designed a long time ago and hasn't been updated much, so at this point it's hard to say exactly what the rationale was, sorry. My *guess* is because including the # (and requiring that the user preserve it while editing) would add a bit of extra complexity, and it's not really necessary when the field is typically labeled "hex".

There's not currently a way to make the field include the # (though you can customize the label if that helps). What was your reasoning for wanting to show it?

**Reply by hzhu** [reply]
Thanks @ecraig12345 for the response, really appreciate it 🙏 .

> What was your reasoning for wanting to show it?

The ability for the HEX input to handle a number `(#)` would help improve usability by allowing users to copy HEX values from popular tools (web browsers, design tools) and paste it directly into the HEX input of the color picker.

Copying a HEX value with `#` from web browser:
![2020-11-30 15 55 28](https://user-images.githubusercontent.com/1811365/100679577-a2e9c280-3324-11eb-8cf3-67142bb91f17.gif)

Copying a HEX value with `#` from design tools:
![2020-11-30 15 52 59](https://user-images.githubusercontent.com/1811365/100679576-a2512c00-3324-11eb-995d-d6c22843fbce.gif)

**Reply by JustSlone** [reply]
Interesting, so then this could be simply solved by ColorPicker allowing a # value to be input into the control, even if the control immediately strips it off. 

Like this:
![colorpicker](https://user-images.githubusercontent.com/40304143/100914721-4a333a80-3488-11eb-9e6a-2a6dca306aa3.gif)

**Reply by hzhu** [reply]
@JustSlone thanks for the response! 

Oh yeah, that would solve it 👍 . However, I should have mentioned that the ability to copy + paste the other direction would also help improve usability. For example, copying a HEX value from the color picker, then pasting it into a web browser (or design tool):

![2020-12-02 11 08 11](https://user-images.githubusercontent.com/1811365/100919527-c4ff5400-348e-11eb-940b-6cadb6d30445.gif)

I imagine (as @ecraig12345 had mentioned) the ability for the HEX input to handle number sign (`#`) and non-number sign values could introduce some complexity to the code. Is the improved usability enough to warrant further engineering and design exploration?

**Comment by ValentinaKozlova** [comment]
Hello @hzhu, preview version of ColorPicker V9 is available:
https://react.fluentui.dev/?path=/docs/preview-components-colorpicker--docs
You might be interested in checking this out
